File: src/main/java/com/cse3063f25grp1/orchestrator/RagOrchestrator.java
package com.cse3063f25grp1.orchestrator;

import com.cse3063f25grp1.config.Config;
import com.cse3063f25grp1.context.Context;
import com.cse3063f25grp1.trace.JsonlTraceSink;
import com.cse3063f25grp1.trace.TraceBus;

public class RagOrchestrator {
    private final Context context;
    private final TraceBus traceBus;

    public RagOrchestrator(Context context) {
        this.context = context;
        this.traceBus = new TraceBus();
        System.out.println("RagOrchestrator created.");
    }

    public void run(Config config, String question) {
        // Register JsonlTraceSink for logging
        JsonlTraceSink jsonlSink = new JsonlTraceSink(config.getLogsDirPath());
        traceBus.register(jsonlSink);

        RagPipeline pipeline = new SequentialRagPipeline(config, context, traceBus);
        pipeline.execute();

    }
}

File: iteration2/src/orchestrator/rag_orchestrator.py
from pathlib import Path

from src.config.config import Config
from src.context.context import Context
from src.orchestrator.sequential_rag_pipeline import SequentialRagPipeline
from src.trace.trace_bus import TraceBus
from src.trace.jsonl_trace_sink import JsonlTraceSink


class RagOrchestrator:
    
    def __init__(self, context: Context):
       
        self.__context = context
        self.__trace_bus = TraceBus()
    
    def run(self, config: Config) -> None:
        jsonl_sink = JsonlTraceSink(config.get_logs_dir_path())
        self.__trace_bus.register(jsonl_sink)
        
        pipeline = SequentialRagPipeline(config, self.__context, self.__trace_bus)
        pipeline.execute()
